apps/music/test/marionette/playlist_test.js

```diff
@@ -245,5 +245,37 @@ marionette('Music player playlist', function() {
       assert.equal(PlaylistHelper.songTitle(songs[1]), 'Crash');
     });
 
+    test('Recently added playlist sort order', function() {
+      // start the app so the music files are added to the database.
+      music.launch();
+      music.waitForFirstTile();
+
+      // close the app
+      music.close();
+
+      // add more files.
+      client.fileManager.add([
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/01.ogg'
+        }
+      ]);
+
+      // start it over. It will add the file above.
+      music.launch();
+      music.waitForFirstTile();
+      music.switchToPlaylistsView();
+
+      music.selectPlaylist('Recently added');
+
+      music.waitForSongs(function(songs) {
+        return songs.length >= 6;
+      });
+      var songs = music.songs;
+
+      assert.equal(PlaylistHelper.songIndex(songs[0]), '1');
+      assert.equal(PlaylistHelper.songTitle(songs[0]),
+                   'The Ecuadorian Embassy');
```

Comment:
Is there any way that PlaylistHelper could have a songFilename() method? If so, just testing the filename against the filename you passed to client.fileManager() would make these tests easier to understand and verify. 

As it is, the test depends on the presence of metadata that is not visible in the test itself.  If you can't test for the actual filename, at least add a comment to when you add 01.ogg so that a reader knows that this song has this title.